### PR TITLE
Convert repo to GitHub actions CI 

### PR DESCRIPTION
**Pull request by kentdr**
The build is still a bit flaky in that both the Windows tests and the Linux tests will pass but usually not in the same build. So we don't want to merge this just yet until it's more stable but we're submitting it for review to get feedback and ideas as to what might be causing the instability.

One theory for the flakiness is that our existing tests go against a local Rabbit environment (either a local install or a local Docker container). Since the GitHub Actions build works against an Azure Container Instance, the theory is that latency is causing intermittent errors.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+      - release-[0-5].*
+  pull_request_target:
+    branches-ignore:
+      - release-[0-5].*
+  workflow_dispatch:
+env:
+  DOTNET_NOLOGO: true
+jobs:
+  build:
+    environment:
```
Comment (bording) [inline .github/workflows/ci.yml:19]: Why did you add the environment? We aren't currently using these for anything because they aren't useful enough to replace Octopus, so it would probably be a good idea to remove this.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+          echo "Tagging container image"
+          $dateTag = "Created=$(Get-Date -Format "yyyy-MM-dd")"
+
+          $ignore = az tag create --resource-id $details.id --tags Package=RabbitMQTransport RunnerOS=${{ runner.os }} $dateTag
+
+          echo "RabbitMQTransport_ConnectionString=host=$fqdn" | Out-File -FilePath $Env:GITHUB_ENV -Encoding utf-8 -Append
+      - name: Run Windows tests
+        if: matrix.name == 'Windows'
+        # -m:1 parameter prevents test projects from being run in parallel, which could cause conflicts since PessimisticLocks project shares same tests
```
Comment (bording) [inline .github/workflows/ci.yml:112]: "PessimisticLocks" aren't a thing in this repo, so this comment isn't valid.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        run: dotnet test src --configuration Release --no-build --logger "GitHubActions;report-warnings=false" -m:1
+      - name: Run Linux tests
+        if: matrix.name == 'Linux'
+        # -m:1 parameter prevents test projects from being run in parallel, which could cause conflicts since PessimisticLocks project shares same tests
```
Comment (bording) [inline .github/workflows/ci.yml:116]: Same with this comment.

File: src/NServiceBus.Transport.RabbitMQ.Tests/NServiceBus.Transport.RabbitMQ.Tests.csproj [hunk 4]
```diff
@@ -7,11 +7,22 @@
     <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
   </PropertyGroup>
 
+  <ItemGroup>
+    <None Remove="myp12.p12" />
```
Comment (bording) [inline src/NServiceBus.Transport.RabbitMQ.Tests/NServiceBus.Transport.RabbitMQ.Tests.csproj:11]: Assuming the path change for the file stays in the PR, then this should be updated as well:

```suggestion
    <None Update="myp12.p12" CopyToOutputDirectory="PreserveNewest" />
```

File: src/NServiceBus.Transport.RabbitMQ.Tests/NServiceBus.Transport.RabbitMQ.Tests.csproj [hunk 5]
```diff
@@ -7,11 +7,22 @@
...
+  </ItemGroup>
+
+  <ItemGroup>
```
Comment (bording) [inline src/NServiceBus.Transport.RabbitMQ.Tests/NServiceBus.Transport.RabbitMQ.Tests.csproj:14]: Remove this entire ItemGroup.

File: src/NServiceBus.Transport.RabbitMQ.Tests/ConnectionString/ConnectionConfigurationTests.cs [hunk 6]
```diff
@@ -9,7 +9,7 @@ public class ConnectionConfigurationTests
     {
         static readonly string connectionString =
             "virtualHost=Copa;username=Copa;host=192.168.1.1:1234;password=abc_xyz;port=12345;requestedHeartbeat=3;" +
-            $"retryDelay=01:02:03;useTls=true;certPath=..{Path.DirectorySeparatorChar}..{Path.DirectorySeparatorChar}..{Path.DirectorySeparatorChar}myp12.p12;certPassPhrase=abc123";
+            $"retryDelay=01:02:03;useTls=true;certPath=.{Path.DirectorySeparatorChar}myp12.p12;certPassPhrase=abc123";
```
Comment (bording) [inline src/NServiceBus.Transport.RabbitMQ.Tests/ConnectionString/ConnectionConfigurationTests.cs:12]: Why was this change required?

File: src/NServiceBus.Transport.RabbitMQ.Tests/ConnectionString/ConnectionConfigurationTests.cs [hunk 7]
```diff
@@ -19,6 +19,7 @@ RabbitMQTransport CreateTransportDefinition(string connectionString)
         [Test]
         public void Should_correctly_parse_full_connection_string()
         {
+            Console.WriteLine($"Current working directory: {Directory.GetCurrentDirectory()}");
```
Comment (bording) [inline src/NServiceBus.Transport.RabbitMQ.Tests/ConnectionString/ConnectionConfigurationTests.cs:22]: Debug stuff that should be removed, right?

File: src/NServiceBus.Transport.RabbitMQ.Tests/ConnectionString/ConnectionConfigurationTests.cs [hunk 8]
```diff
@@ -129,7 +130,8 @@ public void Should_parse_use_tls()
         [Test]
         public void Should_parse_the_cert_path()
         {
-            var connectionConfiguration = CreateTransportDefinition($"host=localhost;certPath=..{Path.DirectorySeparatorChar}..{Path.DirectorySeparatorChar}..{Path.DirectorySeparatorChar}myp12.p12;certPassphrase=abc123");
+            Console.WriteLine($"Current working directory {Directory.GetCurrentDirectory()}");
```
Comment (bording) [inline src/NServiceBus.Transport.RabbitMQ.Tests/ConnectionString/ConnectionConfigurationTests.cs:133]: Debug stuff that should be removed, right?

**Review by bording — CHANGES_REQUESTED** [review]
In addition to the changes mentioned, you're also missing a release.yml workflow.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (kentdr) [reply .github/workflows/ci.yml:19]: I removed the environment section.  It was a holdover from the forked branch.

Comment (bording) [reply .github/workflows/ci.yml:19]: You say you removed it, but it's still here.

File: src/NServiceBus.Transport.RabbitMQ.Tests/NServiceBus.Transport.RabbitMQ.Tests.csproj [hunk 9]
```diff
@@ -20,5 +21,4 @@
     <PackageReference Include="PublicApiGenerator" Version="10.2.0" />
     <PackageReference Include="RabbitMQ.Client" Version="6.2.2" />
   </ItemGroup>
-
```
Comment (bording) [inline src/NServiceBus.Transport.RabbitMQ.Tests/NServiceBus.Transport.RabbitMQ.Tests.csproj:23]: nit; This line should not be removed.

File: .github/workflows/ci.yml [hunk 10]
```diff
@@ -0,0 +1,141 @@
...
+            if(!$response.status)
+            {
+              Write-Output "No response, retrying..."
+              Start-Sleep -m 1000
+            }
+          } until (($response.status) -or ($tries -ge 3))
+
+          if($response.status -ne "ok")
+          {
+            Write-Output "Failed to connect after 3 attampts";
```
Comment (bording) [inline .github/workflows/ci.yml:124]:
```suggestion
            Write-Output "Failed to connect after 3 attempts";
```

File: .github/workflows/ci.yml [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+          if(!$jsonResult)
+          {
+            Write-Output "Failed to create RabbitMQ container"
+            exit 1;
+          }
+
+          $details = $jsonResult | ConvertFrom-Json
+
+          if (!$details.ipAddress) {
+            Write-Output "Failed to create Rabbit container $hostname in $region"
```
Comment (bording) [inline .github/workflows/ci.yml:92]:
```suggestion
            Write-Output "Failed to create RabbitMQ container $hostname in $region"
```

**Review by bording — CHANGES_REQUESTED** [review]
